substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/snippets/ReflectionPlugins.java

```diff
@@ -193,6 +200,53 @@ public boolean apply(GraphBuilderContext b, ResolvedJavaMethod targetMethod, Rec
         });
     }
 
+    /**
+     * For some methods check if folding an invocation using reflection, i.e., by executing the
+     * target method and capturing the result, has undesired side effects, such as triggering
+     * initialization of classes that should be initialized at run time. This is based on knowledge
+     * about the reflection API methods implementation.
+     */
```

Comment:
@cstancu Is this still an issue now that the new class initialization strategy ( https://github.com/oracle/graal/pull/4684) is being used by default? 

This change seems to significantly increase the number of metadata in the image heap.